### ISSUE TITLE
Support for StrictHostKeyChecking

### ISSUE THREAD

**Issue by MattDavisRV** [alebeck/boring#15]
I ran across this package a while ago and was setting up a new computer for work and decided to give it a try.  I'm running into an issue and I've tried searching around, but couldn't find a solution.  

When I attempt to open a connection by running `boring o mydb`, I receive the following output:

```[13:01:35] ERROR Could not open tunnel 'mydb': could not setup SSH client: ssh: handshake failed: knownhosts: key is unknown```

In `~/.bording.toml`, I have this definition:

```
[[tunnels]]
name = "mydb"
local = "3307"
remote = "my-rds-host-name.rds.amazonaws.com:3306"
host = "my-ec2-host-name.amazonaws.com"
user = "ec2-user"
identity = "~/.ssh/my-key-name.pem"
```
I've been using the following command, which does work, to establish the tunnel:

```
ssh -N -L 3307:my-rds-host-name.rds.amazonaws.com:3306 -i ~/.ssh/my-key-name.pem ec2-user@my-ec2-host-name.amazonaws.com
```

In `~/.ssh/config`, I do have an entry for this domain:

```
Host *.amazonaws.com
    StrictHostKeyChecking no
    KexAlgorithms diffie-hellman-group-exchange-sha256
    Ciphers aes256-ctr,aes192-ctr,aes128-ctr
    UserKnownHostsFile /dev/null
```

Given that the ssh command does work, I'm not really sure what else I can debug, do you have any suggestions?

**Comment by alebeck** [comment]
Hi, currently we don’t support the „StrictHostKeyChecking no“ config option yet, so I assume it tried to read host keys from /dev/null, which fails. Can you try again with host key verification? The reason it still works with the ssh command is that we‘re not shelling out to ssh but use low(er) level ssh primitives from golang.org/x/crypto/ssh, so the feature set is not always exactly the same (though that’s the goal).

**Comment by MattDavisRV** [comment]
Thanks for the quick response, I just removed that line and started a new terminal session.  I'm getting the same error when attempting to connect.  I also tried to remove the UserKnownHostFile entry as well, leaving only the Kex and Cipher lines for the host, with the same issue.

**Comment by alebeck** [comment]
Hmm, I guess because now there’s no key entry for that host? Can you connect again using ssh so that the key is added and then try boring open again?

**Comment by MattDavisRV** [comment]
Ah, didn't even think of that.  After removing both of the entries it works now.  If I have some time, I'll try to put in a PR with a new config option to disable to host check in the next few days

**Comment by alebeck** [comment]
So I assume it’s working now? I’m actually currently in the process of adding ProxyJump which needs some refactorings, so I could also include StrictHostKeyChecking while I’m at it. How fast do you need that?

**Comment by MattDavisRV** [comment]
Yes, everything works now with this app! but if I trigger it via the command line I end up with issues, so I'm stuck in limbo at the moment.  

I'm not in a huge rush, but I do have about 10 different ec2 hosts I tunnel into for various applications, it would be nice to be able to manage these in this tool opposed to different alias' that I have setup now.

**Comment by alebeck** [comment]
Alright that’s good! By „trigger it via the command line“ you mean ssh or boring? If the latter, what’s the problem/error messages?

I assume all public keys of all instances are in your knwon_hosts? Generally host key verification should be used, but I get that there are scenarios (particularity in CI/CD and with dynamically allocated instances) where it would disrupt the process too much…

**Comment by MattDavisRV** [comment]
Sorry, I should have more clear here.  When I run the ssh command I posted in my original message directly I do get issues about known_host validation issues.  

These EC2 instances spin up/down each day, so I do need to ignore the hosts and not rely on my known_host file unfortunately.  I don't like the way this works, but it isn't something I can change due to policy.

I'll likely just add the option to skip host key checks in my ssh commands as I migrate over to use boring more.

**Comment by alebeck** [comment]
I see, that makes sense! I’ll keep you updated about the development.

**Comment by alebeck** [comment]
Hi @MattDavisRV, boring 0.11.0 now supports "StrictHostKeyChecking no".
